tests/Integration/PackagePrivate/DatabaseTermIdsAcquirerTest.php

```diff
@@ -0,0 +1,285 @@
+			]
+		];
+
+		$acquiredTermIds = $dbTermIdsAcquirer->acquireTermIds( $termsArray );
+
+		$this->assertSame(
+			3,
+			$this->db->selectRowCount( 'wbt_text', '*' )
+		);
+	}
+
+	public function testAcquireTermIdsStoresOnlyUniqueTextInLang() {
+		$typeIdsAcquirer = new InMemoryTypeIdsAcquirer();
+
+		$dbTermIdsAcquirer = new DatabaseTermIdsAcquirer(
+			$this->db,
+			$this->db,
+			$typeIdsAcquirer
+		);
+
+		$termsArray = [
+			'label' => [
+				'en' => 'same',
+				'de' => 'same',
+			],
+			'description' => [ 'en' => 'same' ],
+			'alias' => [
+				'en' => [ 'same', 'same', 'another', 'yet another' ]
+			]
+		];
+
+		$acquiredTermIds = $dbTermIdsAcquirer->acquireTermIds( $termsArray );
+
+		$this->assertSame(
+			4,
+			$this->db->selectRowCount( 'wbt_text_in_lang', '*' )
+		);
+	}
+
+	public function testAcquireTermIdsStoresOnlyUniqueTermInLang() {
+		$typeIdsAcquirer = new InMemoryTypeIdsAcquirer();
+
+		$dbTermIdsAcquirer = new DatabaseTermIdsAcquirer(
+			$this->db,
+			$this->db,
+			$typeIdsAcquirer
+		);
+
+		$termsArray = [
+			'label' => [
+				'en' => 'same',
+				'de' => 'same',
+			],
+			'description' => [ 'en' => 'same' ],
+			'alias' => [
+				'en' => [ 'same', 'same', 'another', 'yet another' ]
+			]
+		];
+
+		$acquiredTermIds = $dbTermIdsAcquirer->acquireTermIds( $termsArray );
+
+		$this->assertSame(
+			6,
+			$this->db->selectRowCount( 'wbt_term_in_lang', '*' )
+		);
+	}
+
+	public function testAcquireTermIdsReusesExistingTerms() {
+		$termsArray = [
+			'label' => [
+				'en' => 'same',
+				'de' => 'same',
+			],
+			'description' => [ 'en' => 'same' ],
+			'alias' => [
+				'en' => [ 'same', 'same', 'another', 'yet another' ]
+			]
+		];
+
+		// We will populate DB with two terms that both have
+		// text "same". One is of type "label" in language "en",
+		// and the other is of type "alias" in language "en.
+		//
+		// TermIdsAcquirer should then reuse those terms for the given
+		// termsArray above, meaning thoese pre-inserted terms will
+		// appear (their ids) in the returned array from
+		// TermIdsAcquirer::acquireTermIds( $termsArray )
+		$typeIdsAcquirer = new InMemoryTypeIdsAcquirer();
+		$alreadyAcquiredTypeIds = $typeIdsAcquirer->acquireTypeIds(
+			[ 'label', 'description', 'alias' ]
+		);
+
+		$this->db->insert( 'wbt_text', [ 'wbx_text' => 'same' ] );
+		$sameTextId = $this->db->insertId();
+
+		$this->db->insert(
+			'wbt_text_in_lang',
+			[ 'wbxl_text_id' => $sameTextId, 'wbxl_language' => 'en' ]
+		);
+		$enSameTextInLangId = $this->db->insertId();
+
+		$this->db->insert(
+			'wbt_term_in_lang',
+			[ 'wbtl_text_in_lang_id' => $enSameTextInLangId,
+			  'wbtl_type_id' => $alreadyAcquiredTypeIds['label'] ]
+		);
+		$labelEnSameTermInLangId = (string)$this->db->insertId();
+
+		$this->db->insert(
+			'wbt_term_in_lang',
+			[ 'wbtl_text_in_lang_id' => $enSameTextInLangId,
+			  'wbtl_type_id' => $alreadyAcquiredTypeIds['alias'] ]
+		);
+		$aliasEnSameTermInLangId = (string)$this->db->insertId();
```

Comment:
because the interface expects string ids only (Note 2 https://github.com/wmde/mediawiki-term-store/pull/11/files#diff-8b29a0bea7b8ed10130868f7ba1dc7cdR75)